src/components/posts/buttons/ButtonAction.js

```diff
@@ -0,0 +1,27 @@
+function ButtonAction({
```

Comment:
At the moment this isn't really a generic component, by forcing all of these different inputs it isn't leveraging the similar style of the two functions. You could change it up so that it only passes in the `buttonFunction` with the inputs so that you can simplify this component. The other way is to change the inputs so that there is a general `buttonTitle` input, the `id` input, and then maybe a list of extra inputs or something like that. Currently your function inputs are acting as a switch statement when the generic aspect of them could be doing that for you instead